Source/SwiftLintBuiltInRules/Rules/Style/NonFinalClassRule.swift

```diff
@@ -0,0 +1,51 @@
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+@SwiftSyntaxRule(correctable: true, optIn: true)
+struct NonFinalClassRule: Rule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "non_final_class",
+        name: "Non-Final Class",
+        description: "Classes should be marked as `final` unless they are explicitly `open`.",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            Example("final class MyClass {}"),
+            Example("open class MyClass {}"),
+            Example("public final class MyClass {}"),
+        ],
+        triggeringExamples: [
+            Example("class MyClass {}"),
```

Comment:
Please add violation markers `↓` to indicate the exact positions where the violations are expected to appear.